azure-pipelines/pull-request-validation/build-consumers.yml

```diff
@@ -0,0 +1,196 @@
+      commonBranch: $[ dependencies.setupBranch.outputs['setvarStep.commonBranch'] ]  # map in the variable
+    steps:
+    - template: ../templates/steps/automation-cert.yml
+    - checkout: msal
+      displayName: Checkout msal repository
+      clean: true
+      submodules: recursive
+      persistCredentials: True
+    - task: CmdLine@2
+      displayName: Checkout common submodule $(commonBranch)
+      inputs:
+        script: |
+          git fetch
+          git checkout $(commonBranch)
+          git pull
+          git status
+          git rev-parse HEAD
+        workingDirectory: $(Agent.BuildDirectory)/s/common
+    - task: Gradle@2
+      displayName: Assemble msal
+      inputs:
+        tasks: clean msal:assembleLocal
+    - task: Gradle@2
+      displayName: Run msal Unit tests
+      inputs:
+        tasks: msal:testLocalDebugUnitTest -Plabtest -ProbolectricSdkVersion=${{variables.robolectricSdkVersion}}
+  # broker
+  - job: brokerValidation
+    displayName: Broker
+    dependsOn: setupBranch
+    variables:
+      commonBranch: $[ dependencies.setupBranch.outputs['setvarStep.commonBranch'] ]  # map in the variable
+    steps:
+    - template: ../templates/steps/automation-cert.yml
+    - checkout: broker
+      displayName: Checkout broker repository
+      clean: true
+      submodules: recursive
+      persistCredentials: True
+    - task: CmdLine@2
+      displayName: Checkout common submodule $(commonBranch)
+      inputs:
+        script: |
+          git fetch
+          git checkout $(commonBranch)
+          git pull
+          git status
+          git rev-parse HEAD
+        workingDirectory: $(Agent.BuildDirectory)/s/common
+    - task: PowerShell@2
+      displayName: Install Flatbuffers Compiler
+      enabled: False
+      inputs:
+        targetType: inline
+        script: >-
+          Write-Output "Downloading flatbuffers"
+
+          $downloadLink = "https://github.com/google/flatbuffers/releases/download/v1.12.0/flatc_windows.zip"
+
+          Write-Output "Using download link: $downloadLink"
+
+          $dest = "$env:Temp\FlatBufferInstaller.zip"
+
+          (New-Object Net.WebClient).DownloadFile($downloadLink, $dest)
+
+          Write-Output "Extracting flatbuffers archive on machine...."
+
+          Expand-Archive -Path $dest -DestinationPath .\flatc-extracted
+
+          Write-Output "Finished extracting flatbuffers archive on machine...."
+
+          Write-Output "Run flatc"
+
+          .\flatc-extracted\flatc
+
+          echo '##vso[task.setvariable variable=path]$(PATH):$(CWD)\flatc-extracted\'
+
+          flatc
+    - task: Gradle@2
+      displayName: Assemble broker
+      inputs:
+        tasks: clean AADAuthenticator:assembleLocal
+    - task: Gradle@2
+      displayName: Run broker Unit tests
+      inputs:
+        tasks: AADAuthenticator:testLocalDebugUnitTest -Plabtest -ProbolectricSdkVersion=${{variables.robolectricSdkVersion}}
+  # adal
+  - job: adalValidation
```

Comment:
Do these jobs run in parallel or are they sequential?

Reply:
meaning does adalValidation have to wait for msalValidation and brokerValidation to finish or can these be run in parallel?

Reply:
yes jobs runs in parallel, that's why the consumers jobs (adal, msal, broker) depend on the first job `  - job: setupBranch` to finish, once this is done the rest will start in parallel
Example:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/c6860f3f97280704e0a6a3d0b9513e674a5c1cef/azure-pipelines/pull-request-validation/build-consumers.yml#L79